### PR TITLE
upower: Move battery_threshold setting to fu-engine

### PR DESCRIPTION
**Pull request by djcampello**

### CODE REVIEW

File: plugins/powerd/fu-plugin-powerd.c [hunk 1]
```diff
@@ -57,6 +62,28 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 			     g_dbus_proxy_get_name (data->proxy));
 		return FALSE;
 	}
+
+	battery_str = fu_plugin_get_config_value (plugin, "BatteryThreshold");
+	if (battery_str == NULL) {
```
Comment (hughsie) [inline plugins/powerd/fu-plugin-powerd.c:67]: Why the config file override?

Comment (djcampello) [reply plugins/powerd/fu-plugin-powerd.c:67]: This was basically copied as is from fu_plugin_upower.c. I can remove if needed

Comment (hughsie) [reply plugins/powerd/fu-plugin-powerd.c:67]: Are you copying from upower plugin from master?

Comment (hughsie) [reply plugins/powerd/fu-plugin-powerd.c:67]: Also, maybe we should move the fu_context_set_battery_threshold() stuff into the daemon? It seems a bit silly to duplicate it.

Comment (bleungatchromium) [reply plugins/powerd/fu-plugin-powerd.c:67]: Hey @djcampello, it looks like the version of fu-plugin-upower.c you've brought this over from doesn't have this latest commit:
https://github.com/fwupd/fwupd/commit/6f831903926dcdf201328fb450dd17c5ef0d8ecf

The most recent one is cleaner.

Comment (djcampello) [reply plugins/powerd/fu-plugin-powerd.c]: I just fixed it on the recent push... Where would you think this should go in the daemon?

**Review by hughsie — APPROVED** [review]

**Pull request merged**